### DISCUSSION TITLE
Does streamlink support importing cookies as a file like youtube-dl?

### DISCUSSION THREAD

**Discussion by chorong761** [streamlink/streamlink#4313] [Q&A] [answered]
I'm trying to import cookies for a site with no plugin support, but I don't know where to find the cookie I need when using --http-cookie. Does streamlink support a cookie file for the while site like youtube-dl does? Or am I doing something wrong? 

Thank You.

**Comment by bastimeyer** [comment, marked as answer]
> I don't know where to find the cookie I need when using --http-cookie

[As described in the docs](https://streamlink.github.io/cli.html#cmdoption-http-cookie), you can set multiple `--http-cookie=key=value` parameters:
```sh
streamlink --http-cookie 'key1=value2' --http-cookie 'key2=value2' URL QUALITY
```

> Does streamlink support a cookie file like youtube-dl does

No, it does not, but you can add a custom config file that gets loaded via `--config` which then includes those `http-cookie` parameters:
https://streamlink.github.io/cli.html#cmdoption-config

If one wants a `--http-cookie-file` or `--http-header-file` parameter for Streamlink's CLI, they can submit a pull request that will then be reviewed.
